modules/imaging_browser/php/imaging_browser.class.inc

```diff
@@ -284,117 +98,119 @@ END";
+        // the resulting array back to revert it to a key/value (i.e.
+        // acquisition protocol ID/scan type) combination.
+        $config             = \NDB_Config::singleton();
+        $toTable_scan_types = $config->getSetting('tblScanTypes');
+        $scan_id_types      = array_flip(
+            array_intersect_key(
+                array_flip($all_scan_types),
+                array_flip($toTable_scan_types)
+            )
         );
 
-        $this->fieldOptions = [
-                               'sites'         => $siteList,
-                               'projects'      => $list_of_projects,
-                               'visitQCStatus' => $visitQCStatus,
-                               'sequenceTypes' => $scan_types,
-                               'pendingNew'    => $pending,
-                               'configLabels'  => $scan_id_types,
-                              ];
+        return [
+                'sites'         => $siteList,
+                'projects'      => $list_of_projects,
+                'visitQCStatus' => $visitQCStatus,
+                'sequenceTypes' => $all_scan_types,
+                'pendingNew'    => $pending,
+                'configLabels'  => $scan_id_types,
+               ];
 
-        $this->tpl_data['backURL'] = $_SERVER['REQUEST_URI'];
     }
+
     /**
-     * Converts this menu filter to an array of the form
-     *
-     * Headers => (string array)
-     * Data => (array of arrays of row data)
+     * Gets the data source for this menu filter.
      *
-     * @note overloaded function
-     *         Overloading this method to create a list of sessionID that
-     *         will be used for the Navigation Links in  the viewSession
-     *         page.
+     * @return \LORIS\Data\Provisioner
+     */
+    public function getBaseDataProvisioner() : \LORIS\Data\Provisioner
+    {
+        return new ImagingBrowserRowProvisioner();
+    }
+
+    /**
+     * Return a data provisioner of the same type as BaseDataProvisioner, with
+     * additional LORIS filters applied.
      *
-     * @return array
+     * @return \LORIS\Data\Provisioner a provisioner with specific filters added
      */
-    function toArray()
+    public function getDataProvisionerWithFilters(): \LORIS\Data\Provisioner
     {
-        $data  = parent::toArray();
-        $index = array_search('SessionID', $data['Headers']);
-        if ($index !== false) {
-            $_SESSION['State']->setProperty(
-                'mriSessionsListed',
-                array_column($data['Data'], $index)
-            );
+
+        $provisioner = parent::getDataProvisionerWithFilters();
+
+        $user = \User::singleton();
+
+        // UserSiteMatch for imaging browser specific permissions
+        if ($user->hasPermission("imaging_browser_view_allsites") == false) {
+            if ($user->hasPermission("imaging_browser_view_site") == false) {
+                // user only has access to phantom scans from all sites
```

Comment:
there should be a check to see if the user have `imaging_browser_phantom_ownsite`